(GEOS 3.8) Geos function mismatch when building osgEarthSymbology/GEOS.cpp

I am receiving the error shown below when building with release 2.10.1. Geos version installed is 3.8.0-r1. I have replicated this on two different computer systems, both running Gentoo. Have you seen any errors similar to this? 

I appreciate your comments!

Error message:
`[ 28%] Building CXX object src/osgEarthSymbology/CMakeFiles/osgEarthSymbology.dir/GeometryRasterizer.cpp.o
cd /var/tmp/portage/sci-geosciences/osgearth-2.10.1/work/osgearth-2.10.1_build/src/osgEarthSymbology && /usr/bin/x86_64-pc-linux-gnu-g++ -DOSGEARTHSYMBOLOGY_LIBRARY -DOSGEARTH_CXX11 -DOSGEARTH_HAVE_GEOS -DosgEarthSymbology_EXPORTS -I/var/tmp/portage/sci-geosciences/osgearth-2.10.1/work/osgearth-osgearth-2.10.1/src -I/var/tmp/portage/sci-geosciences/osgearth-2.10.1/work/osgearth-osgearth-2.10.1   -DNDEBUG -march=native -O2 -pipe -fPIC   -std=gnu++11 -o CMakeFiles/osgEarthSymbology.dir/GeometryRasterizer.cpp.o -c /var/tmp/portage/sci-geosciences/osgearth-2.10.1/work/osgearth-osgearth-2.10.1/src/osgEarthSymbology/GeometryRasterizer.cpp
/var/tmp/portage/sci-geosciences/osgearth-2.10.1/work/osgearth-osgearth-2.10.1/src/osgEarthSymbology/GEOS.cpp: In function ‘geos::geom::CoordinateSequence* {anonymous}::vec3dArray2CoordSeq(const osgEarth::Symbology::Geometry*, bool, const geos::geom::CoordinateSequenceFactory*)’:
/var/tmp/portage/sci-geosciences/osgearth-2.10.1/work/osgearth-osgearth-2.10.1/src/osgEarthSymbology/GEOS.cpp:67:56: error: cannot convert ‘std::unique_ptr<geos::geom::CoordinateSequence>’ to ‘geos::geom::CoordinateSequence*’ in initialization
   67 |         geom::CoordinateSequence* seq = factory->create( coords );
      |                                         ~~~~~~~~~~~~~~~^~~~~~~~~~
      |                                                        |
      |                                                        std::unique_ptr<geos::geom::CoordinateSequence>
/var/tmp/portage/sci-geosciences/osgearth-2.10.1/work/osgearth-osgearth-2.10.1/src/osgEarthSymbology/GEOS.cpp: In function ‘geos::geom::Geometry* {anonymous}::import(const osgEarth::Symbology::Geometry*, const geos::geom::GeometryFactory*)’:
/var/tmp/portage/sci-geosciences/osgearth-2.10.1/work/osgearth-osgearth-2.10.1/src/osgEarthSymbology/GEOS.cpp:158:65: error: no matching function for call to ‘geos::geom::GeometryFactory::createPolygon(geos::geom::LinearRing*&, std::vector<geos::geom::Geometry*>*&) const’
`

--- a/src/osgEarth/GEOS.cpp
+++ b/src/osgEarth/GEOS.cpp
@@ -63,9 +63,13 @@ namespace
         {
             coords->push_back( coords->front() );
         }
+#if GEOS_VERSION_AT_LEAST(3,8)
+        geom::CoordinateSequence::Ptr seq = factory->create(coords);
+        return seq.get();
+#else
         geom::CoordinateSequence* seq = factory->create( coords );
-
         return seq;
+#endif
     }
 
     geom::Geometry*
@@ -141,12 +145,20 @@ namespace
                     if ( shell )
                     {
                         const Polygon* poly = static_cast<const Polygon*>(input);
+#if GEOS_VERSION_AT_LEAST(3,8)
+                        std::vector<geom::LinearRing*>* holes = poly->getHoles().size() > 0 ? new std::vector<geom::LinearRing*>() : 0L;
+#else
                         std::vector<geom::Geometry*>* holes = poly->getHoles().size() > 0 ? new std::vector<geom::Geometry*>() : 0L;
+#endif
                         if (holes)
                         {
                             for( RingCollection::const_iterator r = poly->getHoles().begin(); r != poly->getHoles().end(); ++r )
                             {
-                                geom::Geometry* hole = import( r->get(), f );
+#if GEOS_VERSION_AT_LEAST(3,8)
+                                geom::LinearRing* hole = dynamic_cast<geom::LinearRing*>(import( r->get(), f ));
+#else
+                                geom::Geometry* hole = import(r->get(), f);
+#endif
                                 if (hole)
                                 {
                                     if (hole->getGeometryTypeId() == geos::geom::GEOS_LINEARRING)

